### PR TITLE
Fail on unknown manifest keys + Add `[tool]` section for 3rd-party tools

### PR DESCRIPTION
**Pull request by tingerrr**
This PR denies unknown fields in both the package manifest and package section of the manifest, ensuring typos are caught by CI. To still allow external configuration, 3rd-party tools can use an entry in the `[tool]` section of the document.

This change is backwards compatible with all existing packages, one notable packages came up during testing, `drafting:0.1.2`, which already used `[tool]` without my prior knowledge.

### CODE REVIEW

**Comment by tingerrr** [comment]
I switched to an empty `Tool` struct, so we retain `Eq` for `PackageManifest`, keeping the deletions of the diff at a minimum.

File: README.md [hunk 1]
```diff
@@ -46,6 +46,17 @@ Optional:
   otherwise unnecessarily increase the bundle size. Don't exclude the README or
   the LICENSE.
 
+Thrid-party tools can add their own entry under the `[tool]` section to attach
```
Comment (laurmaedje) [inline README.md:49]:
```suggestion
Third-party tools can add their own entry under the `[tool]` section to attach
```

File: README.md [hunk 2]
```diff
@@ -46,6 +46,17 @@ Optional:
...
+their own typst specific config to the manifest.
```
Comment (laurmaedje) [inline README.md:50]:
```suggestion
their own Typst-specific configuration to the manifest.
```

**Comment by laurmaedje** [comment]
Thank you!

**Pull request merged**